watch_service.go

```diff
@@ -277,39 +251,39 @@ func (w *Watcher) Watch(listener WatchListener) error {
 		case <-w.watchCTX.Done():
 			return w.watchCTX.Err()
 
-		case ch <- &latest:
+		case ch <- latest:
 		}
 	}
 
 	return nil
 }
 
-func (ws *watchService) fileWatcher(projectName, repoName string, query *Query) (*Watcher, error) {
+func (ws *watchService) fileWatcher(ctx context.Context, projectName, repoName string, query *Query) (*Watcher, error) {
+	return ws.fileWatcherWithTimeout(ctx, projectName, repoName, query, defaultWatchTimeout)
+}
+
+func (ws *watchService) fileWatcherWithTimeout(ctx context.Context, projectName, repoName string, query *Query, timeout time.Duration) (*Watcher, error) {
 	if query == nil {
 		return nil, ErrQueryMustBeSet
 	}
 
-	w := newWatcher(projectName, repoName, query.Path)
-	w.doWatchFunc = func(lastKnownRevision int) <-chan *WatchResult {
-		return ws.watchFile(w.watchCTX, projectName, repoName, strconv.Itoa(lastKnownRevision),
-			query, watchTimeout)
-	}
-	w.convertingResultFunc = func(result *WatchResult) *Latest {
-		value := result.Entry.Content
-		return &Latest{Revision: result.Commit.Revision, Value: value}
+	w := newWatcher(ctx, projectName, repoName, query.Path)
+	w.doWatchFunc = func(ctx context.Context, lastKnownRevision int) *WatchResult {
+		return ws.watchFile(ctx, projectName, repoName, strconv.Itoa(lastKnownRevision),
+			query, timeout)
 	}
 	return w, nil
 }
 
-func (ws *watchService) repoWatcher(projectName, repoName, pathPattern string) (*Watcher, error) {
-	w := newWatcher(projectName, repoName, pathPattern)
-	w.doWatchFunc = func(lastKnownRevision int) <-chan *WatchResult {
-		return ws.watchRepo(w.watchCTX, projectName, repoName, strconv.Itoa(lastKnownRevision),
-			pathPattern, watchTimeout)
-	}
-	w.convertingResultFunc = func(result *WatchResult) *Latest {
-		revision := result.Commit.Revision
-		return &Latest{Revision: revision, Value: revision}
+func (ws *watchService) repoWatcher(ctx context.Context, projectName, repoName, pathPattern string) (*Watcher, error) {
+	return ws.repoWatcherWithTimeout(ctx, projectName, repoName, pathPattern, defaultWatchTimeout)
+}
+
+func (ws *watchService) repoWatcherWithTimeout(ctx context.Context, projectName, repoName, pathPattern string, timeout time.Duration) (*Watcher, error) {
```

Comment:
This line is too long? What's the limit that Golang usually uses? I used 112 characters for Java.

Reply:
Golang have no limit. It's IDE I think. so I will shorten it.